provider/testdata/data/baked/calendar/japanese_v1_u_ca.rs

```diff
@@ -1,10 +1,13 @@
 // @generated
-type DataStruct = & 'static < :: icu_calendar :: provider :: JapaneseErasV1Marker as :: icu_provider :: DataMarker > :: Yokeable ;
-pub static DATA: &[(&str, DataStruct)] = &[
-    ("und-u-ca-japanese", UND_U_CA_JAPANESE),
-    ("und-u-ca-japanext", UND_U_CA_JAPANEXT),
-];
-static UND_U_CA_JAPANESE: DataStruct = &::icu_calendar::provider::JapaneseErasV1 {
+type DataStruct =
+    <::icu_calendar::provider::JapaneseErasV1Marker as ::icu_provider::DataMarker>::Yokeable;
+pub static DATA: litemap::LiteMap<&str, &DataStruct, &[(&str, &DataStruct)]> = unsafe {
+    litemap::LiteMap::from_slice_unchecked(&[
```

Comment:
Nit: I'm not convinced this should be `unsafe`. If we don't have things in the correct sort order, we get GIGO behavior, but not unsafe behavior. See previous discussion in https://github.com/unicode-org/icu4x/issues/1433

(I realize this was probably a comment for the previous PR)

Reply:
Fair, I'll change it

Reply:
How about we name the functions `from_softed_tuple_vec` and `from_sorted_tuple_slice`

Reply:
I would keep the `_unchecked` because we're still not checking the sortedness invariant.

Reply:
Readability-wise, `_sorted` is a pretty strong signal that we are expecting that particular invariant. Also adding `_unchecked` raises questions about whether there are more invariants, which there are not.

If we could use `Ord` in `const` functions then this would be a non-issue because we could just return a `Result`.

Reply:
Yeah but we can't...

I'd expect `from_slice` to accept any slice, and `from_sorted_slice` to accept only sorted slices and *error otherwise*. `from_sorted_slice_unchecked` is what'd I'd expect to do GIGO

Reply:
Should we consider special casing &str and &u8 keys into their own functions where we check the invariant without calling Ord?

Reply:
I'd rather not. We can't even `unwrap` in `const` at the moment (we'd need a verbose `match`), so I don't think we're gaining much by doing that. Databake already uses a lot of unchecked constructors.